### PR TITLE
qTox-portable: Add version 1.16.3

### PR DESCRIPTION
**Pull request by linsui**

### CODE REVIEW

File: bucket/qtox-portable.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            "if (!(Test-Path \"$persist_dir\\qtox.ini\")) {",
+            "    if (Test-Path \"$env:AppData\\Tox\" -PathType Container) {",
+            "        Copy-Item \"$env:AppData\\Tox\\*\" \"$dir\" -Include '*.ini', '*.db', '*.tox'",
+            "    }",
+            "    if (Test-Path \"$dir\\qtox.ini\") {",
+            "        ((Get-Content \"$dir\\qtox.ini\") -replace 'makeToxPortable=false', 'makeToxPortable=true') | Set-Content \"$dir\\qtox.ini\" -Encoding ASCII",
+            "    }",
+            "    else { Set-Content \"$dir\\qtox.ini\" @('[Advanced]', 'makeToxPortable=true') -Encoding ASCII -Force }",
+            "}",
+            "else { Copy-Item \"$persist_dir\\*\" \"$dir\" -Exclude 'qtox.ini' }"
```
Comment (Ash258) [inline bucket/qtox-portable.json]: Allman and K&R mixing?? 🤢

File: bucket/qtox-portable.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            "url": "https://github.com/qTox/qTox/releases/download/v1.16.3/setup-qtox-x86_64-release.exe#/dl.7z",
+            "hash": "e4a36d89739768afedf539cd049bfc7eb19022a2f9c40e52b50081604b0ab027"
+        },
+        "32bit": {
+            "url": "https://github.com/qTox/qTox/releases/download/v1.16.3/setup-qtox-i686-release.exe#/dl.7z",
+            "hash": "d4a50995124b6663c4aa7398520f023e7a76cba9a179371af5c9e45396b3c551"
+        }
+    },
+    "extract_dir": "bin",
+    "installer": {
```
Comment (Ash258) [inline bucket/qtox-portable.json]: Nonse to keep this in fresh new manifest. If user want to migrate then he should do this manually.

File: bucket/qtox-portable.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        "script": [
+            "if (!(Test-Path \"$persist_dir\\qtox.ini\")) {",
+            "    if (Test-Path \"$env:AppData\\Tox\" -PathType Container) {",
+            "        Copy-Item \"$env:AppData\\Tox\\*\" \"$dir\" -Include '*.ini', '*.db', '*.tox'",
+            "    }",
+            "    if (Test-Path \"$dir\\qtox.ini\") {",
+            "        ((Get-Content \"$dir\\qtox.ini\") -replace 'makeToxPortable=false', 'makeToxPortable=true') | Set-Content \"$dir\\qtox.ini\" -Encoding ASCII",
```
Comment (Ash258) [inline bucket/qtox-portable.json]:
```suggestion
            "        (Get-Content \"$dir\\qtox.ini\") -replace 'makeToxPortable=false', 'makeToxPortable=true' | Set-Content \"$dir\\qtox.ini\" -Encoding ASCII",
```

File: bucket/qtox-portable.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{
+    "version": "1.16.3",
+    "description": "Chat, voice, video, and file transfer IM client using the encrypted peer-to-peer Tox protocol.",
+    "homepage": "https://qtox.github.io/",
```
Comment (Ash258) [inline bucket/qtox-portable.json]:
```suggestion
    "homepage": "https://qtox.github.io",
```

File: bucket/qtox-portable.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            "    if (Test-Path \"$dir\\qtox.ini\") {",
+            "        ((Get-Content \"$dir\\qtox.ini\") -replace 'makeToxPortable=false', 'makeToxPortable=true') | Set-Content \"$dir\\qtox.ini\" -Encoding ASCII",
+            "    }",
+            "    else { Set-Content \"$dir\\qtox.ini\" @('[Advanced]', 'makeToxPortable=true') -Encoding ASCII -Force }",
+            "}",
+            "else { Copy-Item \"$persist_dir\\*\" \"$dir\" -Exclude 'qtox.ini' }"
+        ]
+    },
+    "uninstaller": {
+        "script": "Copy-Item \"$dir\\*\" \"$persist_dir\" -Include '*.ini', '*.db', '*.tox' -Force"
```
Comment (Ash258) [inline bucket/qtox-portable.json]: How many files are there?

File: bucket/qtox-portable.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    "bin": "qtox.exe",
+    "shortcuts": [
+        [
+            "qtox.exe",
+            "qTox"
+        ]
+    ],
+    "persist": "avatars",
+    "checkver": {
+        "github": "https://github.com/qTox/qTox/"
```
Comment (Ash258) [inline bucket/qtox-portable.json]:
```suggestion
        "github": "https://github.com/qTox/qTox"
```

File: bucket/qtox-portable.json [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/qTox/qTox/releases/download/v$version/setup-qtox-x86_64-release.exe#/dl.7z"
```
Comment (Ash258) [inline bucket/qtox-portable.json]: You most likely need to remove $PLUGINSDIR

File: bucket/qtox-portable.json (hunk 5 above)
Comment (linsui) [reply bucket/qtox-portable.json]: Every profile has a username.ini, a username.db,and a username.tox. And a qtox.ini.

File: bucket/qtox-portable.json (hunk 2 above)
Comment (linsui) [reply bucket/qtox-portable.json]: Oh, yeah, you are right.

File: bucket/qtox-portable.json (hunk 7 above)
Comment (linsui) [reply bucket/qtox-portable.json]: There is no $PLUGINSDIR or uninstall.

File: bucket/qtox-portable.json [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            "hash": "d4a50995124b6663c4aa7398520f023e7a76cba9a179371af5c9e45396b3c551"
+        }
+    },
+    "extract_dir": "bin",
+    "installer": {
+        "script": [
+            "if (!(Test-Path \"$persist_dir\\qtox.ini\")) {",
+            "    Set-Content \"$dir\\qtox.ini\" @('[Advanced]', 'makeToxPortable=true', '[%General]', 'checkUpdates=false') -Encoding ASCII -Force",
+            "}",
+            "else { Move-Item \"$persist_dir\\*\" \"$dir\" -Include '*.ini', '*.db', '*.tox' }"
```
Comment (Ash258) [inline bucket/qtox-portable.json]: ...
Move else to L21

**Pull request merged**